core/src/main/java/org/apache/druid/java/util/common/parsers/ObjectFlatteners.java

```diff
@@ -231,6 +235,11 @@ public Map<String, Object> toMap(T obj)
      */
     Function<T, Object> makeJsonQueryExtractor(String expr);
 
+    /**
+     * Create a "field" extractor for nested json expressions
+     */
+    Function<T, Object> makeJsonTreeExtractor(List<String> nodes);
```

Comment:
@jasonk000 @FrankChen021 - this can potentially break extensions developed outside apache repo. While the interface is not marked `ExtensionPoint`, it does get extended outside the core. My suggestion will be to throw an `UnsupportOperationException` in a default implementation. Also `getRootField` needs to be called as before when `fieldSpec.getType()` returns `ROOT`. 
what do you think?

Reply:
Do we need to care about an interface which is not marked as `ExtensionPoint`? Although we can add a default implementation for this interface. Another question is should we mark it as `ExtensionPoint`?

Reply:
It's already implemented in Avro, orc, and parquet extension. So +1 to both of your suggestions. We should add a default impl and and mark it extension point.

Reply:
Hi @jasonk000 , could you make some changes here according to above opinions? I think we're close to get this PR to be merge.

Reply:
OK let me take a look. I'm a little hesitant as I'm not in a position to _run_ it now against trunk+this rebased, but I'll put up the changes and we can take it from there!